c2corg_ui/static/js/documentediting.js

```diff
@@ -0,0 +1,269 @@
+goog.provide('app.DocumentEditingController');
+goog.provide('app.documentEditingDirective');
+
+goog.require('app');
+goog.require('ol.format.GeoJSON');
+goog.require('ol.geom.Point');
+
+
+
+/**
+ * @constructor
+ * @return {angular.Directive} The directive specs.
+ * @ngInject
+ */
+app.documentEditingDirective = function() {
+  return {
+    restrict: 'A',
+    scope: true,
+    controller: 'appDocumentEditingController',
+    controllerAs: 'editCtrl',
+    bindToController: true
+  };
+};
+
+
+app.module.directive('appDocumentEditing', app.documentEditingDirective);
+
+
+
+/**
+ * @param {angular.Scope} $scope Scope.
+ * @param {angular.JQLite} $element Element.
+ * @param {angular.Attributes} $attrs Attributes.
+ * @param {angular.$http} $http
+ * @param {string} apiUrl Base URL of the API.
+ * @constructor
+ * @ngInject
+ * @export
+ */
+app.DocumentEditingController = function($scope, $element, $attrs, $http,
```

Comment:
Yes, the same form is used both for creating a new doc or editing an existing one: the "only" main difference is that the controller makes a request to the API to get the values of the existing doc, which is not needed for a new doc.
The mako template is the same in both cases as well.

I will add such a comment. If relevent I can rename the controller and the form template, but I thought that "editing" made sense, even for creating a doc.